src/Backups/BackupsWorker.cpp

```diff
@@ -454,7 +454,9 @@ void BackupsWorker::doBackup(
             /// Prepare backup entries.
             BackupEntries backup_entries;
             {
-                BackupEntriesCollector backup_entries_collector{backup_query->elements, backup_settings, backup_coordination, backup_create_params.read_settings, context};
+                BackupEntriesCollector backup_entries_collector(
+                    backup_query->elements, backup_settings, backup_coordination,
+                    backup_create_params.read_settings, context, *backups_thread_pool);
```

Comment:
Using `backups_thread_pool` might be not a good idea in case when BackupEntriesCollector runs in the same pool. Imagine it has only one thread, we schedule another task and wait -> deadlock.